.github/workflows/cargo_publish.yml

```diff
@@ -38,9 +39,11 @@ jobs:
 
       - name: Create release tag and check forward commit to main
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # the default secrets.GITHUB_TOKEN won't trigger a workflow run
+          # when tagging, but it will if we provide a custom PAT
+          GITHUB_TOKEN: ${{ secrets.DENOBOT_PAT }}
```

Comment:
See https://github.community/t/pushing-a-tag-within-an-action-doesnt-trigger-another-workflow/201080/5

https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow

> When you use the repository's GITHUB_TOKEN to perform tasks, events triggered by the GITHUB_TOKEN will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs. For example, if a workflow run pushes code using the repository's GITHUB_TOKEN, a new workflow will not run even when the repository contains a workflow configured to run when push events occur.